diff --git a/src/google/protobuf/unittest_custom_options.proto b/src/google/protobuf/unittest_custom_options.proto
index 1812d71..9a2b57a 100644
--- a/src/google/protobuf/unittest_custom_options.proto
+++ b/src/google/protobuf/unittest_custom_options.proto
@@ -304,7 +304,9 @@ message VariousComplexOptions {
 // See descriptor_unittest.cc.
 
 message AggregateMessageSet {
-  option message_set_wire_format = true;
+  // MessageSets are not supported by the Go protobuf implemetation.
+  // We could build with `-tags protolegacy`, but we dont need this.
+  // option message_set_wire_format = true;
 
   extensions 4 to max;
 }
